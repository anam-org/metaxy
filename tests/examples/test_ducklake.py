"""Tests for DuckLake example."""

import subprocess
import sys
from pathlib import Path


def test_ducklake_demo_preview():
    """Test that the DuckLake demo runs and previews attachment SQL."""
    example_dir = Path("examples/src/examples/ducklake")

    result = subprocess.run(
        [sys.executable, "-m", "examples.ducklake.demo"],
        capture_output=True,
        text=True,
        timeout=10,
        cwd=example_dir,
    )

    assert result.returncode == 0, (
        f"Demo failed: {result.stderr}\nstdout: {result.stdout}"
    )
    print(result.stdout)

    assert "DuckLake store initialised" in result.stdout
    assert "Preview of DuckLake ATTACH SQL:" in result.stdout
    assert "INSTALL" in result.stdout or "LOAD" in result.stdout
    assert "ATTACH" in result.stdout or "ducklake" in result.stdout.lower()
