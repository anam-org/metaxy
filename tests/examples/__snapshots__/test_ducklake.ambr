# serializer version: 1
# name: test_ducklake_demo_snapshot
  '''
  DuckLake store initialised. Extensions: ['ducklake', 'hashfuncs']
  
  Preview of DuckLake ATTACH SQL:
     INSTALL ducklake;
     LOAD ducklake;
     CREATE OR REPLACE SECRET secret_ducklake ( TYPE DUCKLAKE, METADATA_PATH '/tmp/ducklake_meta.db', DATA_PATH '/tmp/ducklake_storage' );
     ATTACH 'ducklake:secret_ducklake' AS ducklake (API_VERSION '0.2', OVERRIDE_DATA_PATH true);
     USE ducklake;
  
  '''
# ---
